fl4health/clients/mr_mtl_client.py

```diff
@@ -0,0 +1,188 @@
+from logging import INFO
+from pathlib import Path
+from typing import Dict, List, Optional, Sequence, Tuple
+
+import torch
+import torch.nn as nn
+from flwr.common.logger import log
+from flwr.common.typing import Config, NDArrays, Scalar
+
+from fl4health.checkpointing.checkpointer import TorchCheckpointer
+from fl4health.clients.basic_client import BasicClient
+from fl4health.parameter_exchange.full_exchanger import FullParameterExchanger
+from fl4health.utils.losses import LossMeterType, TrainingLosses
+from fl4health.utils.metrics import Metric
+
+
+class MrMtlClient(BasicClient):
+    def __init__(
+        self,
+        data_path: Path,
+        metrics: Sequence[Metric],
+        device: torch.device,
+        loss_meter_type: LossMeterType = LossMeterType.AVERAGE,
+        checkpointer: Optional[TorchCheckpointer] = None,
+        lam: float = 1.0,
```

Comment:
This isn't part of the original implementation (and shouldn't be in this PR), but it would be interesting to see what effect adapting this value based on the adaptive FedProx approach or the generalization gap of FedDG-GA would have on the approach. A similar idea could be said for the Ditto parameter.

Reply:
I am confused by this comment, which part do you mean exactly?

Reply:
Apologies, this was just me recording a thought about something to investigate in the future. That is, would MR-MTL or Ditto benefit from an adaptive implementation of the FedProx-like parameter $\lambda$. It could be adjusted as they suggested in the original paper, or perhaps we could adjust it based on a similar measure to that proposed in FedDG-GA (which is a generalization gap).

TL;DR: No need to do anything. Was just thinking about potential future experimentation.